st2common/st2common/services/executions.py

```diff
@@ -132,9 +142,12 @@ def _get_parent_execution(child_liveaction_db):
 def update_execution(liveaction_db, publish=True):
     execution = ActionExecution.get(liveaction__id=str(liveaction_db.id))
     decomposed = _decompose_liveaction(liveaction_db)
+    kw = {}
     for k, v in six.iteritems(decomposed):
-        setattr(execution, k, v)
-    execution = ActionExecution.add_or_update(execution, publish=publish)
+        kw['set__' + k] = v
```

Comment:
Because it wasn't particularly clear it is the goal. `liveaction__id` few lines above is also mongo-specific and if you do `ag "\w__\w"`, you soon realize we don't really follow this convention outside of `trace` and `trigger` both written by you. Saying that, I'm ok with that convention and will try to figure out how to rewrite this function to match it, but we need everyone else to understand and comply with it too for it to have any effect.